components/SelectField.tsx

```diff
@@ -0,0 +1,150 @@
+import { FC, useState, useEffect, ReactNode } from 'react'
+
+import {
+  Box,
+  Flex,
+  FlexProps,
+  HTMLChakraProps,
+  Popover,
+  PopoverBody,
+  PopoverContent,
+  PopoverTrigger,
+  useDisclosure,
+  useMultiStyleConfig,
+} from '@chakra-ui/react'
+import { TriangleDownIcon } from '@chakra-ui/icons'
+
+type OptionType = {
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  value: any
+  label: string | ReactNode
+  helperText: string | ReactNode
+}
+
+interface OptionProps extends HTMLChakraProps<'div'> {
+  option: OptionType
+}
+
+const Option: FC<OptionProps> = ({ option, ...props }) => (
+  <Box {...props}>
+    <Box className="select-field_option_label">{option?.label}</Box>
+    <Box className="select-field_option_text">{option?.helperText}</Box>
```

Comment:
Minor, but if we're gonna go this direction, we should use a proper BEM formatting:

```
// .base__element--modifier
.select-field__option-text
```

I also could suggest the library [blem](https://www.npmjs.com/package/blem) for this (I am the maintainer):

```
import blem from 'blem'

const bem = blem('select-field')
// [...]
className={bem('option-label')} // 'select-field__option-label'
className={bem('option-text')} // 'select-field__option-text'
```